ens/utils.py

```diff
@@ -181,3 +183,48 @@ def assert_signer_in_modifier_kwargs(modifier_kwargs):
 
 def is_none_or_zero_address(addr):
     return not addr or addr == '0x' + '00' * 20
+
+
+def resolve_content_record(resolver, name):
+    is_eip1577 = resolver.functions.supportsInterface('0xbc1c58d1').call()
+    is_legacy = resolver.functions.supportsInterface('0xd8389dc5').call()
+
+    namehash = normal_name_to_hash(name)
+
+    if is_eip1577:
+        raw_content_hash = resolver.functions.contenthash(namehash).call().hex()
+
+        if is_none_or_zero_address(raw_content_hash):
+            return None
+
+        decoded_content_hash = content_hash.decode(raw_content_hash)
+        type_content_hash = content_hash.get_codec(raw_content_hash)
+
+        return {
+            'type': type_content_hash,
+            'hash': decoded_content_hash,
+        }
+
+    if is_legacy:
+        content = resolver.functions.content(namehash).call()
+
+        if is_none_or_zero_address(content):
+            return None
```

Comment:
I'd like to get away from this `None` return in favor of using an exception raising pattern.  cc @kclowes as this might need to go through a deprecation cycle in the other functions if this pattern is used there.

For here I think we'd probably want a new exception class for this.

Reply:
Other methods (`address`) also use `None` return.